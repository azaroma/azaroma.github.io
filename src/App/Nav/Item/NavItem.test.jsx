import * as React from "react";
import { shallow } from "enzyme";
import NavItem from ".";

const props = {
  name: "Nav element",
  route: "/"
};

describe("Nav Element", () => {
  const wrapper = shallow(<NavItem {...props} />);

  describe("when provided the correct props", () => {
    it("renders without crashing", () => {
      expect(wrapper.exists()).toBe(true);
      expect(wrapper).toHaveLength(1);
    });

    it("looks as expected", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
