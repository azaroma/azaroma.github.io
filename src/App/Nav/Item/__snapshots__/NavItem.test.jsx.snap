// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Element when provided the correct props looks as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavElement
    name="Nav element"
    route="/"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/"
      >
        Nav element
      </Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Nav element",
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": "Nav element",
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/"
        >
          Nav element
        </Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Nav element",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Nav element",
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
